ggdeploymentd/src/deployment_handler.c

```diff
@@ -2674,14 +2776,63 @@ static void handle_deployment(
         }
     }
 
+    ret = delete_saved_deployment_from_config();
+    if (ret != GGL_ERR_OK) {
+        GGL_LOGE("Failed to delete saved deployment info for bootstrap.");
+    }
+
+    // clear contents of deployed_components for next deployment
+    deployed_components = GGL_KV_VEC((GglKV[64]) { 0 });
+
     *deployment_succeeded = true;
 }
 
 static GglError ggl_deployment_listen(GglDeploymentHandlerThreadArgs *args) {
+    // check for in progress deployment in case of bootstrap
+    GglDeployment bootstrap_deployment = { 0 };
+    GglError ret = retrieve_in_progress_deployment(
+        &bootstrap_deployment, &deployed_components
+    );
+    if (ret != GGL_ERR_OK) {
+        GGL_LOGD("No deployments previously in progress detected.");
+    } else {
+        GGL_LOGD(
+            "Found previously in progress deployment %.*s. Resuming "
+            "deployment.",
+            (int) bootstrap_deployment.deployment_id.len,
+            bootstrap_deployment.deployment_id.data
+        );
+        update_current_jobs_deployment(
+            bootstrap_deployment.deployment_id, GGL_STR("IN_PROGRESS")
+        );
```

Comment:
IoT Jobs Listener may not have the Jobs ID for the deployment ID to update its state on-boot.